### PR TITLE
[Woo POS] Cannot reconnect to Reader after turning internet off and on

### LINKED ISSUE #12497: [Woo POS] Cannot reconnect to Reader after turning internet off and on
**Describe the bug**
A clear and concise description of what the bug is. Please be as descriptive as possible; issues lacking detail, or for any other reason than to report a bug, may be closed without action.

Cannot reconnect to Reader after turning the internet off and on. I tested on a simulator with a simulated card reader, so this issue may be invalid.

**To Reproduce**
1. Open POS
2. Connect to reader
3. Turn off internet
4. Disconnect reader by tapping on floating button
5. Turn on internet
6. Notice card reader button and "Collect payment" doesn't work anymore

**Screenshots**

https://github.com/user-attachments/assets/eda393e7-4189-47ed-aefd-7f0239b6c57e

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active,
  or [Storefront](https://woocommerce.com/products/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device:
- Android version:
- WooCommerce Android version:

**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
</details>

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #12497 

### Description

This PR addresses the issue of handling no internet connectivity when attempting to connect to the card reader.

**Previous Behavior:**
When attempting to connect the card reader without an internet connection, the app would not provide any feedback or response. This was due to the app first checking whether all onboarding conditions were met before initiating the card reader connection. In the case of no internet, the app would navigate to an onboarding screen that explains the need for an internet connection. However, since the onboarding screens were not integrated into the POS mode, the app became unresponsive.

**Updated Behavior:**
With this PR, we now check for internet connectivity before attempting to start the WooPosCardReaderActivity. If no internet connection is detected, the activity is not launched, and instead, a toast message is displayed asking the user to ensure a proper internet connection before proceeding.

### Steps to reproduce

1. Navigate to POS mode (More menu -> POS mode)
2. Click on connect to card reader button
3. While connecting, turn off the internet
4. Cancel the ongoing connection dialog
5. Try to connect to card reader again and notice the button won't respond anymore.

### Testing information

I've tested this on tablet emulator by turning on and off internet.

### Images/gif

**Before**

https://github.com/user-attachments/assets/eda393e7-4189-47ed-aefd-7f0239b6c57e

**After**

https://github.com/user-attachments/assets/66604926-674f-499e-bbfd-2a508f1706cc

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>Toast</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/235">20.4</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -4251,6 +4251,7 @@
     <string name="woopos_floating_toolbar_menu_open_content_description">Toolbar with card reader status. Menu is open. Double tap to interact.</string>
     <string name="woopos_floating_toolbar_pop_up_menu_content_description">Open toolbar menu</string>
     <string name="woopos_floating_toolbar_pop_up_menu_open_content_description">Popup menu with options. Swipe to navigate through items.</string>
+    <string name="woopos_no_internet_message">It looks like you\'re not connected to the internet.\n\nEnsure your Wi-Fi is turned on. If you\'re using mobile data, make sure it\'s enabled in your device settings.</string>
```
Comment (samiuelson) [inline WooCommerce/src/main/res/values/strings.xml]: The string is not rendered correctly, new lines are cut:
Screenshot_20240912_171300

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/WooPosHomeViewModel.kt [hunk 2]
```diff
@@ -21,11 +24,14 @@ class WooPosHomeViewModel @Inject constructor(
         initialValue = WooPosHomeState(
             screenPositionState = WooPosHomeState.ScreenPositionState.Cart.Visible,
             productsInfoDialog = WooPosHomeState.ProductsInfoDialog(isVisible = false),
-            exitConfirmationDialog = WooPosHomeState.ExitConfirmationDialog(isVisible = false)
+            exitConfirmationDialog = WooPosHomeState.ExitConfirmationDialog(isVisible = false),
         )
     )
     val state: StateFlow<WooPosHomeState> = _state
 
+    private val _toastEvent = MutableSharedFlow<Int>()
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/WooPosHomeViewModel.kt]: 💡 I'd improve the readability of the `_toastEvent` prop type. I'd personally wrap the Int with a dedicated data class, e.g. 

```kotlin
data class Toast(
    @StringRes val  message: Int,
)
```

even better could be to emit ready to use string:
```kotlin
data class Toast(
    val  message: String,
)
```

or alternatively
```kotlin
import kotlin.Int as ToastMessageStringRes
private val _toastEvent = MutableSharedFlow<ToastMessageStringRes>()
val toastEvent: SharedFlow<ToastMessageStringRes> = _toastEvent
```

**Review by samiuelson — COMMENTED** [review]
Thanks for the PR, @AnirudhBhat! I added a few improvement ideas. Let me know what you think.

File: WooCommerce/src/main/res/values/strings.xml (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/res/values/strings.xml]: Thanks for the heads up! Fixed

no_internet_toast_message

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/WooPosHomeViewModel.kt (hunk 2 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/WooPosHomeViewModel.kt]: Thanks for the suggestion! Improved it here: https://github.com/woocommerce/woocommerce-android/pull/12569/commits/f49db88ad693dbfb5faafc447f49543da7239fda

**Review by samiuelson — APPROVED** [review]
LGTM :shipit:

**Pull request merged**